Classes/Base/Context.php

```diff
@@ -330,4 +331,46 @@ protected static function addGroups( \Aimeos\MShop\Context\Item\Iface $context )
 
 		return $context;
 	}
+
+
+	/**
+	 * Adds the frontend date time to the context
+	 *
+	 * @param \Aimeos\MShop\Context\Item\Iface $context Context object including config
+	 * @return \Aimeos\MShop\Context\Item\Iface Modified context object
+	 */
+	protected static function addDateTime( \Aimeos\MShop\Context\Item\Iface $context )
+	{
+		// Handle the admin panel, according to the version number.
+		if ( version_compare( TYPO3_version, '9.2.0', '<' )
+            && isset($GLOBALS['TSFE'])
+        ) {
+			// The old admin panel saves it's stuff inside the user settings of
+			// the current admin user. These settings will get used, even if the
+			// actual panel gets deactivated.
+			if ( $GLOBALS['BE_USER']->uc['TSFE_adminConfig']['display_top'] === '1'
```

Comment:
You should do more `isset()` checks for `$GLOBALS['BE_USER']`, `$GLOBALS['BE_USER']->uc`, `$GLOBALS['BE_USER']->uc['TSFE_adminConfig']`, `$GLOBALS['BE_USER']->uc['TSFE_adminConfig']['display_top']` and `$GLOBALS['BE_USER']->uc['TSFE_adminConfig']['preview_simulateDate']`

Reply:
I've done some quick tests (5.5 till 7.3). I was unable to trigger anything when looking with `isset()` deep into a not existing array structure.

Am I missing something here?

Reply:
What condition did you use?

Reply:
```php

error_reporting(E_ALL);
$test = [];
var_dump(isset($test['some']['deep']['stuff']));

```

There is also an example in the documentation:
https://www.php.net/manual/de/function.isset.php#example-6551
at 'Die funktioniert ebenfalls für Arrayelemente: ' a little bit down, the last var_dump().

Reply:
Maybe `$GLOBALS['BE_USER']->uc` behaves a bit differently?

Reply:
I did miss something:
Directly accessing a not existing array part triggers a notice.

Sorry . . .